llvm/lib/Target/SPIRV/SPIRVEmitIntrinsics.cpp

```diff
@@ -255,7 +259,19 @@ Instruction *SPIRVEmitIntrinsics::visitGetElementPtrInst(GetElementPtrInst &I) {
 }
 
 Instruction *SPIRVEmitIntrinsics::visitBitCastInst(BitCastInst &I) {
-  SmallVector<Type *, 2> Types = {I.getType(), I.getOperand(0)->getType()};
+  Value *Source = I.getOperand(0);
+  
+  // SPIR-V, contrary to LLVM 17+ IR, supports bitcasts between pointers of
+  // varying element types. In case of IR coming from older versions of LLVM
+  // such bitcasts do not provide sufficient information, should be just skipped
+  // here, and handled in insertPtrCastInstr.
+  if (I.getType()->isPointerTy()) {
+    I.replaceAllUsesWith(Source);
+    I.eraseFromParent();
+    return &I;
```

Comment:
@iliya-diyachkov Here I am removing the nop bitcasts from older versions of LLVM (ptr -> ptr) and returning orphaned instruction (requiring a check `I->getParent()` at line 575). Alternatively I could return nullptr if it would make more sense.

Reply:
In my opinion, nullptr is easier to understand.

Reply:
If you go the `nullptr` way, maybe it would be time to add a comment on those visitors to explain `Returns the newly created intrinsic, nullptr otherwise`?
It just seem to contradict what `visitSwitchInst` is doing tho.